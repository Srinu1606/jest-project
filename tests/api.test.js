jest.setTimeout(30000);
const request = require("supertest");
const endpoints = require('../config/testData');
const postData = require('../config/postData');




describe("API", () => {

  // test("employees",async () => {
  //   const employeeWise = await request(baseURL)
  //     .post("/optHolidays/opt-holiday-config")
  //     .set("Authorization", `Bearer ${authToken}`)
  //     .expect(201);

     

  // });
  // test("employeess",async () => {
  //   const holidaysResponse = await request(baseURL)
  //     .get("/holidays/all")
  //     .set("Authorization", `Bearer ${authToken}`)
  //     .expect(200);

  //     expect(holidaysResponse.body).toMatchSnapshot()

  // });
  it("Holidays", async () => {
    try {
      const holidaysList = await request(endpoints.name)
        .post(endpoints.API)
        .send(postData.body)
        .set("Authorization", endpoints.auth)
        .set('postman', true);
      
      // 1. Basic Equality Matchers
      expect(holidaysList.status).toBe(201);  // Exact equality (===)
      expect(holidaysList.body.success).toBeTruthy();  // Checks for truthy value
      expect(holidaysList.body.error).toBeFalsy();  // Checks for falsy value
      
      // 2. Object Property Matchers
      expect(holidaysList.body).toHaveProperty('success', true);  // Check property and value
      expect(holidaysList.body).toHaveProperty('years');  // Check property existence
      expect(holidaysList.body).toEqual(expect.objectContaining({  // Check object contains specific properties
        success: true,
        message: expect.any(String)  // Check type of value
      }));
      
      // 3. Array Matchers
      expect(Array.isArray(holidaysList.body.years)).toBe(true);  // Check if it's an array
      expect(holidaysList.body.years).toContain(2024);  // Check array contains value
      expect(holidaysList.body.years).toHaveLength(4);  // Check array length
      expect(holidaysList.body.years).toEqual(expect.arrayContaining([2024, 2025]));  // Check array contains values
      
      // 4. String Matchers
      expect(holidaysList.body.message).toMatch(/successfully/);  // Check string matches regex
      expect(holidaysList.body.message).toContain('successfully');  // Check string contains substring
      
      // 5. Number Matchers
      expect(holidaysList.body.years[0]).toBeGreaterThan(2023);  // Check if number is greater than
      expect(holidaysList.body.years[0]).toBeLessThan(2025);  // Check if number is less than
      
      // 6. Type Checking Matchers
      expect(typeof holidaysList.body.success).toBe('boolean');  // Check if boolean
      expect(typeof holidaysList.body.message).toBe('string');  // Check if string
      expect(Array.isArray(holidaysList.body.years)).toBe(true);  // Check if array
      
      // 7. Snapshot Testing
      expect(holidaysList.body).toMatchSnapshot();  // Compare with saved snapshot
      
    } catch (error) {
      console.error('Test failed:', error.message);
      throw error;
    }
  });
  
});
