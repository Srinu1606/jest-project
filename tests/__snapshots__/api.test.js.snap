// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API Holidays 1`] = `
{
  "data": [
    {
      "_id": "65c60e7c4108ddd23c3b28c5",
      "bulk_upload": 0,
      "cancel_leaves": 1,
      "client": "starwars",
      "createdAt": "2024-02-09T11:37:32.352Z",
      "dateValue": "2024-01-31",
      "disabled": 0,
      "dt": "2024-01-31",
      "employee_type": [],
      "entity_id": 1,
      "floating_holiday": 0,
      "holiday_code": "1_20240209113732_20240131_16",
      "holiday_display_code": "STAR_HOLIDAY_2024_E1",
      "holiday_image": null,
      "holiday_name": "star holiday",
      "holiday_repeat": 0,
      "holiday_type": "general",
      "id": "65c60e7c4108ddd23c3b28c5",
      "link_type": 1,
      "linked_locations": [],
      "migrated": true,
      "opt_location": [],
      "original_date": "2024-01-31",
      "primary_location": [],
      "role_type": 0,
      "rulegrp_ids": [],
      "selectAll": 1,
      "shifts": [
        "WO_1",
        "general_shift_1",
        "shift_1",
        "MAYOO_1",
        "Rest_day_1",
        "9_to_7_1",
      ],
      "updatedAt": "2025-02-03T19:43:29.054Z",
      "work_location": [],
      "year": 2024,
    },
    {
      "_id": "65cb2ec8741cd60fe9fc5e8a",
      "bulk_upload": 0,
      "cancel_leaves": 1,
      "client": "starwars",
      "createdAt": "2024-02-13T08:56:40.106Z",
      "dateValue": "2024-02-10",
      "disabled": 0,
      "dt": "2024-02-10",
      "employee_type": [],
      "entity_id": 1,
      "floating_holiday": 0,
      "holiday_code": "1_20240213085640_20240210_17",
      "holiday_display_code": "FEB_10_2024_E1",
      "holiday_image": null,
      "holiday_name": "feb 10",
      "holiday_repeat": 0,
      "holiday_type": "general",
      "id": "65cb2ec8741cd60fe9fc5e8a",
      "link_type": 0,
      "linked_locations": [
        1,
        4,
        13,
        24,
        14,
        12,
        7,
        15,
        17,
        18,
        16,
        11,
        25,
        26,
        28,
        29,
        27,
        30,
        31,
      ],
      "migrated": true,
      "opt_location": [],
      "original_date": "2024-02-10",
      "primary_location": [],
      "role_type": 0,
      "rulegrp_ids": [],
      "selectAll": 0,
      "shifts": [],
      "updatedAt": "2025-02-03T19:43:29.054Z",
      "work_location": [],
      "year": 2024,
    },
    {
      "_id": "65c5c81ee044be76be726548",
      "bulk_upload": 0,
      "cancel_leaves": 0,
      "client": "starwars",
      "createdAt": "2024-02-09T06:37:18.392Z",
      "dateValue": "2024-02-14",
      "disabled": 0,
      "dt": "2024-02-14",
      "employee_type": [],
      "entity_id": 1,
      "floating_holiday": 0,
      "holiday_code": "1_20240209063718_20240214_9",
      "holiday_display_code": "HOLIDAY_TEST_1_2024_E1",
      "holiday_image": null,
      "holiday_name": "Holiday test 1",
      "holiday_repeat": 0,
      "holiday_type": "general",
      "id": "65c5c81ee044be76be726548",
      "link_type": 0,
      "linked_locations": [
        1,
        4,
        13,
        24,
        14,
        12,
        7,
        15,
        17,
        18,
        16,
        11,
        25,
        26,
        28,
        29,
        27,
        30,
        31,
      ],
      "migrated": true,
      "opt_location": [],
      "original_date": "2024-02-14",
      "primary_location": [],
      "role_type": 0,
      "rulegrp_ids": [],
      "selectAll": 0,
      "shifts": [],
      "updatedAt": "2025-02-03T19:43:29.054Z",
      "work_location": [],
      "year": 2024,
    },
    {
      "_id": "668d46f6ba595e26e2473f3a",
      "bulk_upload": 0,
      "cancel_leaves": 1,
      "client": "starwars",
      "createdAt": "2024-07-09T14:19:34.105Z",
      "dateValue": "2024-07-03",
      "disabled": 0,
      "dt": "2024-07-03",
      "employee_type": [],
      "entity_id": 1,
      "floating_holiday": 0,
      "holiday_code": "1_20240709141934_20240703_174",
      "holiday_display_code": "SHIFT_BASED_CHECK_2024_E1",
      "holiday_image": null,
      "holiday_name": "shift based check",
      "holiday_repeat": 0,
      "holiday_type": "public_holiday",
      "id": "668d46f6ba595e26e2473f3a",
      "link_type": 1,
      "linked_locations": [],
      "migrated": true,
      "opt_location": [],
      "original_date": "2024-07-03",
      "primary_location": [],
      "role_type": 0,
      "rulegrp_ids": [],
      "selectAll": 1,
      "shifts": [
        "WO_1",
        "general_shift_1",
        "shift_1",
        "MAYOO_1",
        "Rest_day_1",
        "9_to_7_1",
        "06001500_1",
      ],
      "updatedAt": "2025-02-03T19:43:29.054Z",
      "work_location": [],
      "year": 2024,
    },
    {
      "_id": "668d4821f0466e7a45c9061f",
      "bulk_upload": 0,
      "cancel_leaves": 1,
      "client": "starwars",
      "createdAt": "2024-07-09T14:24:33.919Z",
      "dateValue": "2024-07-22",
      "disabled": 0,
      "dt": "2024-07-22",
      "employee_type": [],
      "entity_id": 1,
      "floating_holiday": 0,
      "holiday_code": "1_20240709142433_20240722_177",
      "holiday_display_code": "SHIFT_REVISION_2024_E1",
      "holiday_image": null,
      "holiday_name": "shift revision",
      "holiday_repeat": 0,
      "holiday_type": "public_holiday",
      "id": "668d4821f0466e7a45c9061f",
      "link_type": 1,
      "linked_locations": [],
      "migrated": true,
      "opt_location": [],
      "original_date": "2024-07-22",
      "primary_location": [],
      "role_type": 0,
      "rulegrp_ids": [],
      "selectAll": 0,
      "shifts": [
        "general_shift_1",
        "shift_1",
        "MAYOO_1",
        "Rest_day_1",
      ],
      "updatedAt": "2025-02-03T19:43:29.054Z",
      "work_location": [],
      "year": 2024,
    },
    {
      "_id": "66c97f8e9809b2222d1f4e58",
      "bulk_upload": 0,
      "cancel_leaves": 1,
      "client": "starwars",
      "createdAt": "2024-08-24T06:37:02.688Z",
      "dateValue": "2024-08-17",
      "disabled": 0,
      "dt": "2024-08-17",
      "employee_type": [],
      "entity_id": 1,
      "floating_holiday": 0,
      "holiday_code": "1_20240824063702_20240817_231",
      "holiday_display_code": "AUG_17_2024_E1",
      "holiday_image": "holidays_-hLjD05UosJ8z_Q0yg3yr.jpg",
      "holiday_name": "AUg 17",
      "holiday_repeat": 0,
      "holiday_type": "general",
      "id": "66c97f8e9809b2222d1f4e58",
      "link_type": 0,
      "linked_locations": [
        1,
        4,
        13,
        24,
        14,
        12,
        7,
        15,
        17,
        18,
        16,
        11,
        25,
        26,
        28,
        29,
        27,
        30,
        31,
        48,
      ],
      "migrated": true,
      "opt_location": [],
      "optional_holiday_type": "",
      "original_date": "2024-08-17",
      "primary_location": [],
      "role_type": 0,
      "rulegrp_ids": [],
      "selectAll": 0,
      "shifts": [],
      "updatedAt": "2025-02-03T19:43:29.054Z",
      "work_location": [],
      "year": 2024,
    },
    {
      "_id": "66cc4f2785c4c85bad25597c",
      "bulk_upload": 0,
      "cancel_leaves": 1,
      "client": "starwars",
      "createdAt": "2024-08-26T09:47:19.056Z",
      "dateValue": "2024-08-19",
      "disabled": 0,
      "dt": "2024-08-19",
      "employee_type": [],
      "entity_id": 1,
      "floating_holiday": 0,
      "holiday_code": "1_20240826094719_20240819_254",
      "holiday_display_code": "OPTIONAL_HOLIDAY_2024_E1",
      "holiday_image": null,
      "holiday_name": "optional holiday",
      "holiday_repeat": 0,
      "holiday_type": "general",
      "id": "66cc4f2785c4c85bad25597c",
      "link_type": 0,
      "linked_locations": [
        1,
        4,
        13,
        24,
        14,
        12,
        53,
        7,
        15,
        17,
        18,
        16,
        11,
        25,
        26,
        28,
        29,
        27,
        30,
        31,
        48,
      ],
      "migrated": true,
      "opt_location": [],
      "optional_holiday_type": "",
      "original_date": "2024-08-19",
      "primary_location": [],
      "role_type": 0,
      "rulegrp_ids": [],
      "selectAll": 0,
      "shifts": [],
      "updatedAt": "2025-02-03T19:43:29.054Z",
      "work_location": [],
      "year": 2024,
    },
    {
      "_id": "66cd75007d5a4ae1961175a6",
      "bulk_upload": 0,
      "cancel_leaves": 0,
      "client": "starwars",
      "createdAt": "2024-08-27T06:41:04.948Z",
      "dateValue": "2024-08-20",
      "disabled": 0,
      "dt": "2024-08-20",
      "employee_type": [],
      "entity_id": 1,
      "floating_holiday": 0,
      "holiday_code": "1_20240820_20240827064104_255",
      "holiday_display_code": "LEAVE_HOLIDAYS_1_2024_E1",
      "holiday_name": "Leave Holidays 1",
      "holiday_repeat": 0,
      "holiday_type": "general",
      "id": "66cd75007d5a4ae1961175a6",
      "link_type": 0,
      "linked_locations": [
        1,
        4,
        13,
        24,
        14,
        12,
        53,
      ],
      "migrated": true,
      "opt_location": [],
      "original_date": "2024-08-20",
      "primary_location": [],
      "role_type": 0,
      "rulegrp_ids": [],
      "shifts": [],
      "updatedAt": "2025-02-03T19:43:29.054Z",
      "work_location": [],
      "year": 2024,
    },
    {
      "_id": "66c7f11f1aa5ba805e0509d8",
      "bulk_upload": 0,
      "cancel_leaves": 1,
      "client": "starwars",
      "createdAt": "2024-08-23T02:17:03.403Z",
      "dateValue": "2024-08-24",
      "disabled": 0,
      "dt": "2024-08-24",
      "employee_type": [],
      "entity_id": 1,
      "floating_holiday": 0,
      "holiday_code": "1_20240823021703_20240824_227",
      "holiday_display_code": "AUG_24_2024_E1",
      "holiday_image": null,
      "holiday_name": "aug 24",
      "holiday_repeat": 0,
      "holiday_type": "general",
      "id": "66c7f11f1aa5ba805e0509d8",
      "link_type": 0,
      "linked_locations": [
        1,
        4,
        13,
        24,
        14,
        12,
        7,
        15,
        17,
        18,
        16,
        11,
        25,
        26,
        28,
        29,
        27,
        30,
        31,
        48,
      ],
      "migrated": true,
      "opt_location": [],
      "optional_holiday_type": "",
      "original_date": "2024-08-24",
      "primary_location": [],
      "role_type": 0,
      "rulegrp_ids": [],
      "selectAll": 0,
      "shifts": [],
      "updatedAt": "2025-02-03T19:43:29.054Z",
      "work_location": [],
      "year": 2024,
    },
    {
      "_id": "66cd75007d5a4ae1961175a7",
      "bulk_upload": 0,
      "cancel_leaves": 0,
      "client": "starwars",
      "createdAt": "2024-08-27T06:41:04.949Z",
      "dateValue": "2024-08-28",
      "disabled": 0,
      "dt": "2024-08-28",
      "employee_type": [],
      "entity_id": 1,
      "floating_holiday": 0,
      "holiday_code": "1_20240828_20240827064104_256",
      "holiday_display_code": "LEAVE_HOLIDAYS_2_2024_E1",
      "holiday_name": "Leave Holidays 2",
      "holiday_repeat": 0,
      "holiday_type": "national_holiday",
      "id": "66cd75007d5a4ae1961175a7",
      "link_type": 0,
      "linked_locations": [
        1,
        4,
        13,
        24,
        14,
        12,
        53,
      ],
      "migrated": true,
      "opt_location": [],
      "original_date": "2024-08-28",
      "primary_location": [],
      "role_type": 0,
      "rulegrp_ids": [],
      "shifts": [],
      "updatedAt": "2025-02-03T19:43:29.054Z",
      "work_location": [],
      "year": 2024,
    },
    {
      "_id": "66cd75007d5a4ae1961175a8",
      "bulk_upload": 0,
      "cancel_leaves": 0,
      "client": "starwars",
      "createdAt": "2024-08-27T06:41:04.949Z",
      "dateValue": "2024-08-30",
      "disabled": 0,
      "dt": "2024-08-30",
      "employee_type": [],
      "entity_id": 1,
      "floating_holiday": 0,
      "holiday_code": "1_20240830_20240827064104_257",
      "holiday_display_code": "LEAVE_HOLIDAYS_3_2024_E1",
      "holiday_name": "Leave Holidays 3",
      "holiday_repeat": 0,
      "holiday_type": "public_holiday",
      "id": "66cd75007d5a4ae1961175a8",
      "link_type": 0,
      "linked_locations": [
        1,
        4,
        13,
        24,
        14,
        12,
        53,
      ],
      "migrated": true,
      "opt_location": [],
      "original_date": "2024-08-30",
      "primary_location": [],
      "role_type": 0,
      "rulegrp_ids": [],
      "shifts": [],
      "updatedAt": "2025-02-03T19:43:29.054Z",
      "work_location": [],
      "year": 2024,
    },
    {
      "_id": "66cdab30d41651dd9efdd0e8",
      "bulk_upload": 0,
      "cancel_leaves": 1,
      "client": "starwars",
      "createdAt": "2024-08-27T10:32:16.608Z",
      "dateValue": "2024-08-31",
      "disabled": 0,
      "dt": "2024-08-31",
      "employee_type": [],
      "entity_id": 1,
      "floating_holiday": 0,
      "holiday_code": "1_20240827103216_20240831_258",
      "holiday_display_code": "HOLIDAY_CHECK_2024_E1",
      "holiday_image": null,
      "holiday_name": "holiday check",
      "holiday_repeat": 0,
      "holiday_type": "general",
      "id": "66cdab30d41651dd9efdd0e8",
      "link_type": 0,
      "linked_locations": [
        1,
        4,
        13,
        24,
        14,
        12,
        53,
        7,
        15,
        17,
        18,
        16,
        11,
        25,
        26,
        28,
        29,
        27,
        30,
        31,
        48,
      ],
      "migrated": true,
      "opt_location": [],
      "optional_holiday_type": "",
      "original_date": "2024-08-31",
      "primary_location": [],
      "role_type": 0,
      "rulegrp_ids": [],
      "selectAll": 0,
      "shifts": [],
      "updatedAt": "2025-02-03T19:43:29.054Z",
      "work_location": [],
      "year": 2024,
    },
    {
      "_id": "66f47b632d9946cf82a3ea03",
      "bulk_upload": 0,
      "cancel_leaves": 1,
      "client": "starwars",
      "createdAt": "2024-09-25T21:06:43.067Z",
      "dateValue": "2024-09-01",
      "disabled": 0,
      "dt": "2024-09-01",
      "employee_type": [],
      "entity_id": 1,
      "floating_holiday": 0,
      "holiday_code": "1_20240925210643_20240901_268",
      "holiday_display_code": "SEP_1_2024_E1",
      "holiday_image": "holidays_fkyhEd5iSHyavmg3lK48I.PNG",
      "holiday_name": "sep 1",
      "holiday_repeat": 0,
      "holiday_type": "general",
      "id": "66f47b632d9946cf82a3ea03",
      "link_type": 0,
      "linked_locations": [
        1,
        4,
        13,
        24,
        14,
        12,
        53,
        48,
      ],
      "migrated": true,
      "opt_location": [],
      "optional_holiday_type": "",
      "original_date": "2024-09-01",
      "primary_location": [],
      "role_type": 0,
      "rulegrp_ids": [],
      "selectAll": 0,
      "shifts": [],
      "updatedAt": "2025-02-03T19:43:29.054Z",
      "work_location": [],
      "year": 2024,
    },
    {
      "_id": "6752c2b615f6f0b78a9c10f0",
      "bulk_upload": 0,
      "cancel_leaves": 0,
      "client": "starwars",
      "createdAt": "2024-12-06T09:24:06.252Z",
      "dateValue": "2024-11-18",
      "disabled": 0,
      "dt": "2024-11-18",
      "employee_type": [],
      "entity_id": 1,
      "floating_holiday": 0,
      "holiday_code": "1_20241206092406_20241118_290",
      "holiday_display_code": "NOV_18_2024_E1",
      "holiday_image": null,
      "holiday_name": "nov 18",
      "holiday_repeat": 0,
      "holiday_type": "optional",
      "id": "6752c2b615f6f0b78a9c10f0",
      "link_type": 5,
      "linked_locations": [],
      "migrated": true,
      "opt_location": [],
      "optional_holiday_type": "Work Location",
      "original_date": "2024-11-18",
      "primary_location": [],
      "role_type": 0,
      "rulegrp_ids": [],
      "selectAll": 0,
      "shifts": [],
      "updatedAt": "2025-02-03T19:43:29.054Z",
      "work_location": [
        1,
        2,
        4,
        3,
      ],
      "year": 2024,
    },
  ],
  "error": false,
  "message": "Fetched holidays successfully",
  "success": true,
  "years": [
    2024,
    2025,
    2026,
    2027,
  ],
}
`;
